examples/plot_fpca.py

```diff
@@ -25,7 +25,19 @@
 # to explore datasets and obtain conclusions about said dataset using this
 # technique.
 #
-# First we are going to fetch the Berkeley Growth Study data. This dataset
+# FPCA is a dimensionality reduction method for functional data that aims to 
+# reduce the complexity of studying observations by finding a finite number of
+# principal components, which are the directions that capture the main modes 
+# of variation across the function (the most important directions in which the
+# curves vary). FPCA can be though of as particular case of a basis expansion 
+# that uses K components for a fixed K, with the particularity being that the 
```

Comment:
I actually liked very much your sentence "among all basis expansions that use K components for a fixed K, the FPC expansion explains most of the variation in X.". Why did you remove it?

I also noted that your commit messages are not very informative. If you want, you may try to use `git rebase` in interactive mode to rewrite your git history and squash all your commits together (or in other situations, to squash some of them, change the messages or reorder them). It is not strictly necessary (I could squash them when merging, or leave them) but it is a tool that is sometimes useful and you may want to try.

Note that, in general, rewriting git history should only be done in private branches that you share with nobody, or with a few other developers you communicate with (so, never rebase master or develop). After rebasing you necessarily have to push with the "--force" (or better "--force-with-lease") option, which means that any other developer that is using the branch would have it in an incompatible state and should reset it.